libs/priceactionlab/CMakeLists.txt

```diff
@@ -78,4 +78,14 @@ target_link_libraries(${LIB_NAME}_unit_tests PRIVATE testinfra)
 enable_testing()
 add_test(NAME ${LIB_NAME}_unit_tests COMMAND ${LIB_NAME}_unit_tests)
 
+# Set the working directory for the test to the project root
+set_tests_properties(${LIB_NAME}_unit_tests
+    PROPERTIES
+        WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}"
+)
+
+# This line for DATASET_FILES might be redundant or could be removed if
+# setting the working directory solves the issue for all dataset access needs.
+# If ${DATASET_FILES} is defined elsewhere and needed for other purposes, it can remain.
```

Comment:
The sole purpose of dataset right now is to be accessible.
My main worry with putting the tests working directory to the root is that we could have tests messing up with files

Reply:
I was just trying to get the test to run. Please let me know your recommendation here and I will follow it.

Reply:
Not sure here but maybe we need to move some of the `.txt` from this sub directory directly into the `dataset` directory above? I can have a stab at it later